src/lib/Util/PermissionUtil.php

```diff
@@ -18,8 +141,14 @@ class PermissionUtil
      *
      * @return array
      */
-    public function flattenArrayOfLimitations(array $hasAccess): array
+    private function flattenArrayOfLimitationsForCurrentUser(array $hasAccess): array
```

Comment:
If you go as far as changing class name please leave original `PermissionUtil` in place to avoid breaking changes (here you are changing method visibility).